lesson1/assignment/yours.sol

```diff
@@ -0,0 +1,77 @@
+pragma solidity ^0.4.14;
+
+/** @title Payroll contract. */
+contract Payroll {
+
+    uint employeeSalary;
+    address employeeWallet;
+    uint constant payDuration = 10 seconds;
+    uint lastPayday = now;
+    
+    /** @dev Initialize Payroll contract.
+      * @param salary Salary of the employee.
+      * @param wallet Wallet address of the employee.
+      */
+    function Payroll(uint salary, address wallet) {
+        employeeSalary = salary * 1 wei;
+        employeeWallet = wallet;
+    }
+    
+    /** @dev set salary.
+      * @param salary Width of the rectangle.
+      */
+    function setSalary(uint salary) {
+        employeeSalary = salary * 1 wei;
+    }
+    
+    /** @dev set employee wallet address.
+      * @param wallet Wallet address of the employee.
+      */
+    function setWalletAddress(address wallet) {
+        employeeWallet = wallet;
+    }
+
+    /** @dev add fund to employer.
+      * @return employer balance after adding fund.
+      */
+    function addFund() payable returns (uint) {
+        return this.balance;
+    }
+    
+    /** @dev calculate how many times the employer can pay the employee
+      * @return the times that employee can get paid from employer's balance
+      */
+    function calculateRunway() returns (uint) {
+        if (employeeSalary == 0) {
+            revert();
```

Comment:
Good that you thought about this. Luckily, EVM throws upon division by zero. Refer to this thread.
https://github.com/ethereum/solidity/issues/670

Reply:
Thanks for your advice.